launcher/src/main/java/org/apache/spark/launcher/LauncherConnection.java

```diff
@@ -95,15 +95,15 @@ protected synchronized void send(Message msg) throws IOException {
   }
 
   @Override
-  public void close() throws IOException {
+  public synchronized void close() throws IOException {
```

Comment:
do we still need to change this method?

Reply:
We never *needed* to change it, but the extra code wasn't doing anything useful, so I chose the simpler version.